tests/trainer/test_data_loading.py

```diff
@@ -117,7 +117,7 @@ def on_train_start(self, *_) -> None:
 def test_replace_sampler_raises():
     trainer = Trainer()
     with pytest.raises(ValueError, match="needs to subclass `torch.utils.data.DataLoader"):
-        trainer.replace_sampler(object(), object(), mode="fit")  # noqa
+        trainer.replace_sampler(object(), object(), mode="fit")
```

Comment:
pycharm will complain about this line since the input type is not the expected.

the noqa should stay.

There are other false negatives, mostly for tests which are testing error raising situations

Reply:
then we shall add error type

Reply:
![Screenshot 2021-07-27 at 15 32 32](https://user-images.githubusercontent.com/11067892/127162658-ba7f35bd-3544-4087-9dcd-0ed84b2ebbf2.png)

What is the error type for this one?

The problem is that different checkers report different error types, there doesn't seem to be a standardization of error codes

Reply:
ok, can PyCharm has another mark for skipping this issue?
but it is just pycharm, or also another tool/checker is complaining?

Reply:
and this is the only one, right?

Reply:
> can PyCharm has another mark for skipping this issue?

I don't know, but `noqa` works across tools.

> or also another tool/checker is complaining?

other tools may complain, for example mypy

> and this is the only one, right?

Other `noqa`s removed by this PR have the same issue

Reply:
I see all check passing: flake8, mypy, pre-commit, ... so to me only PyCharm has an issue...

Reply:
we don't run `mypy` on the test directory, otherwise it would.

see this example

```python
$ cat kk.py 
def foo(x: int) -> None:
    print(x)


foo(5)

foo(object())
$ python kk.py 
5
<object object at 0x1034a44d0>
$ mypy kk.py 
kk.py:7: error: Argument 1 to "foo" has incompatible type "object"; expected "int"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
$ flake8 kk.py 
```

(flake8 does not complain)

Reply:
yes, but mypy has own command for ignore which we use elsewhere in codebase `# type: ignore`

Reply:
btw, if there were only about 5 issues in the test folder (about 5 were removed) why we did not run mypy on all codebase?

Reply:
We don't run mypy for the tests folder